### ISSUE TITLE
Fix Compilation on OSX

### ISSUE THREAD

**Issue by apavlo** [cmu-db/peloton#504]
I can't test this. Somebody else should give it a try.

**Comment by seojungmin** [comment]
I think we better leave this issue until the postgresql parser port is fully done. The first problem I found was in the parser. These are the actions to be done:

(i) There is a type bug around opt_index_type in the parser.   And, the current parser requires Bison > 3.0, due to the %code directive, so if we going to keep the current one, the installation script for OS X should include this. 
(ii) Macros for htobe16/32/64 should be added. RUSAGE_THREAD macro is not defined in OS X for getrusage().  Platform.h would be a good place, but we may need some rule to suit these.
(iii) We may have to revise all the C++ warnings coming out from clang in OS X. With the current cmake configuration, -Werror with clang makes me crazy. :-) It whines on and on. Nothing very significant, but it is worth looking at them sometime. 

I only completed 14% of the build process, so there may be more to be fixed.
